File: kotlin-workshop/part-2-light-mocks/src/main/kotlin/gateway/PaymentGateway.kt
package gateway

import domain.PaymentRequest
import domain.PaymentResult

/**
 * Gateway interface for payment processing.
 * Injected into services - easy to mock!
 */
interface PaymentGateway {
    fun process(request: PaymentRequest): PaymentResult
    fun refund(transactionId: String): Boolean
}

File: csharp-workshop/Part1.StaticDeps/Infrastructure/PaymentGateway.cs
using Part1.StaticDeps.Domain;

namespace Part1.StaticDeps.Infrastructure;

/// <summary>
/// Static payment gateway — requires static mocking (JustMock/Typemock) to test.
/// Java equivalent requires PowerMock.mockStatic().
/// </summary>
public static class PaymentGateway
{
    public static PaymentResult ProcessPayment(string customerId, double amount)
    {
        if (amount > 0)
        {
            return new PaymentResult("", PaymentStatus.Success, Guid.NewGuid().ToString());
        }
        return new PaymentResult("", PaymentStatus.Failed, null);
    }

    public static bool Refund(string transactionId)
    {
        return true;
    }
}
